hack/dev-build.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+###############################################################################
+#
+# dev-build.sh
+#
+# Convenience script for developing kops AND nodeup.
+#
+# This script (by design) will handle building a full kops cluster in AWS,
+# with a custom version of the Nodeup binary compiled at runtime.
+#
+# Example usage
+#
+# S3_BUCKET="s3://my-dev-s3-state \
+# CLUSTER_DOMAIN="mydomain.io" \
+# CLUSTER_PREFIX="prefix" \
+# NODEUP_BUCKET="s3-devel-bucket-name-store-nodeup" \
+# ./dev-build.sh
+#
+###############################################################################
+
+# This script assumes it's in $KOPS/hack
+
+KOPS_DIRECTORY="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+[ -z "$S3_BUCKET" ] && echo "Need to set S3_BUCKET" && exit 1;
+[ -z "$CLUSTER_DOMAIN" ] && echo "Need to set CLUSTER_DOMAIN" && exit 1;
+[ -z "$CLUSTER_PREFIX" ] && echo "Need to set CLUSTER_PREFIX" && exit 1;
+[ -z "$NODEUP_BUCKET" ] && echo "Need to set NODEUP_BUCKET" && exit 1;
```

Comment:
I am going to RFC you guys..

But when naming variables we should use `uri` `urn` and `url` where appropriate

`NODEUP_BUCKET`   ->   `NODEUP_BUCKET_URN=nodeup-bucket-name`
`NODEUP_BUCKER_URL=s3://nodeup-bucket-name`

+1 to using `KOPS_STATE_STORE` instead of `S3_BUCKET`

But whatever we change these to (and I know it's annoying because of the hard coded env.vars being checked throughout the code base) we just need to make sure they are meaningful to newcomers..

I would rather see `NODEUP_S3_LOCATION_NO_PROTOCOL` instead of `NODEUP_BUCKET` - as at least that way we know what to set it it ;) 

http://www.ietf.org/rfc/rfc3986.txt
https://prateekvjoshi.files.wordpress.com/2014/02/uri-vs-url-vs-urn.jpg